tools/configure/configure.ml

```diff
@@ -900,45 +299,59 @@ let find_suffix prefix path = match prefix with
      else
        Absolute path
 
-let env_prefix =
-  match prefs.prefix with
-  | None ->
-    begin
-      try Some (Sys.getenv "COQ_CONFIGURE_PREFIX")
-      with
-      | Not_found when prefs.interactive -> None
-      | Not_found -> Some Sys.(getcwd () ^ "/../install/default")
-    end
-  | p -> p
-
-let do_one_instdir (var,msg,uservalue,selfcontainedlayout,unixlayout) =
-  let dir,suffix =
-    match uservalue, env_prefix with
-    | Some d, _ -> d,find_suffix env_prefix d
-    | None, Some p ->
-      let suffix = if arch_is_win32 then selfcontainedlayout else relativize unixlayout in
-      use_suffix p suffix, suffix
-    | None, None ->
-      let suffix = if unix then unixlayout else selfcontainedlayout in
-      let base = if unix then "/usr/local" else "C:/coq" in
-      let dflt = use_suffix base suffix in
-      let () = printf "Where should I install %s [%s]? " msg dflt in
-      let line = read_line () in
-      if line = "" then (dflt,suffix) else (line,find_suffix None line)
+(* This computes the actual effective path for an install directory,
+   based on the given prefix; if prefix is absent, it is assumed that
+   the profile is "local" *)
+let do_one_instdir ~prefix ~arch InstallDir.{var; msg; uservalue; selfcontainedlayout; unixlayout} =
+  (var,msg),
+  match uservalue, prefix with
+  | Some d, p -> d, find_suffix p d
+  | None, Some p ->
+    let suffix = if (arch_is_win32 arch) then selfcontainedlayout else relativize unixlayout in
+    use_suffix p suffix, suffix
+  | None, None ->
+    let suffix = if (unix arch) then unixlayout else selfcontainedlayout in
+    let base = if (unix arch) then "/usr/local" else "C:/coq" in
+    let dflt = use_suffix base suffix in
+    let () = printf "Where should I install %s [%s]? " msg dflt in
+    let line = read_line () in
+    if line = "" then (dflt,suffix) else (line,find_suffix None line)
+
+let install_dirs prefs arch =
+  let prefix =
+    match prefs.prefix with
+    | None ->
+      begin
+        try Some (Sys.getenv "COQ_CONFIGURE_PREFIX")
+        with
+        | Not_found when prefs.interactive -> None
+        | Not_found -> Some Sys.(getcwd () ^ "/../install/default")
```

Comment:
Don't we have a directory type that can be used? Why use strings here?

Reply:
Do we have that? I am not aware.

Reply:
Paths are strings in base OCaml AFAICT.

Reply:
Ok, I seem to recall seeing `/` being used to deliminate path concat.

Reply:
Oh indeed, we had `let ( / ) = Filename.concat` there, that could be a good idea, but I'm not sure what was the thing with windows paths when we added this auto-check.

Reply:
I'm keeping this for now, as it is currently the same in `master`; but hopefully we can indeed rework that whole code setup soon.